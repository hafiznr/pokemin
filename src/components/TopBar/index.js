import { Box } from "@chakra-ui/react"
import { useHistory, useLocation } from "react-router";

import ChevronLeftIcon from "../CustomIcon/ChevronLeftIcon";

const TopBar = () => {
  const history = useHistory();
  const location = useLocation();
  const { pathname } = location;
  const pathnameArr = pathname.split("/");

  const menus = {
    pokemon: 'Detail',
    mypokemon: 'My Pokémon'
  };

  return (
    <Box 
      bgColor="red.500"
      color="white"
      fontWeight="bold"
      padding="8px"
      textAlign="center"
      maxWidth="480px"
    >
      {history?.length && (
        <Box
          pos="absolute"
          top="8px"
          cursor="pointer"
          onClick={() => history.goBack()}
          data-testid="back-button"
        >
          <ChevronLeftIcon />
        </Box>
      )}
      <Box lineHeight="24px" fontSize="14px">
        {menus[pathnameArr[1]] || 'Pokémin'}
      </Box>
    </Box>
  )
}

export default TopBar;