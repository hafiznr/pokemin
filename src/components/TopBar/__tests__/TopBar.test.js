import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import TopBar from '../';

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/"
  }),
  useHistory: () => ({
    length: 2,
    push: ()=> {}
  }),
}));

test('renders TopBar', () => {
  const { container } = render(
    <BrowserRouter>
      <TopBar/>
    </BrowserRouter>
  );

  expect(container).toMatchSnapshot();
  fireEvent.click(screen.getByTestId('back-button'));
});