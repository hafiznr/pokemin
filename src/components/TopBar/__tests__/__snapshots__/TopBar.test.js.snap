// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TopBar 1`] = `
<div>
  <div
    class="css-1p24ppz"
  >
    <div
      class="css-1y41ry6"
      data-testid="back-button"
    >
      <svg
        fill="none"
        height="24"
        style="display: inline-block;"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M16.2426 6.34317L14.8284 4.92896L7.75739 12L14.8285 19.0711L16.2427 17.6569L10.5858 12L16.2426 6.34317Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="css-1qbkptb"
    >
      Pok√©min
    </div>
  </div>
</div>
`;
